### PR TITLE
*: support reading historial data for infoschema v2

### LINKED ISSUE #51679: support snapshot infoschema for historial read and stale read

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve?

Issue Number: close #51679, ref #50959

Problem Summary:

### What changed and how does it work?

Use schema ts instead of `math.MaxUint64` for infoschemaV2

### Check List

Tests 

- [X] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/infoschema/builder.go [hunk 1]
```diff
@@ -547,31 +546,6 @@ func (b *Builder) applyDropSchema(diff *model.SchemaDiff) []int64 {
 	return tableIDs
 }
 
-func (b *Builder) applyDropTableV2(diff *model.SchemaDiff, dbInfo *model.DBInfo, tableID int64, affected []int64) []int64 {
```
Comment (tiancaiamao) [inline pkg/infoschema/builder.go:550]: This function is moved to file infoschema_v2.go, no change.

File: pkg/infoschema/builder.go [hunk 2]
```diff
@@ -655,7 +629,7 @@ func (b *Builder) buildAllocsForCreateTable(tp model.ActionType, dbInfo *model.D
 	return autoid.NewAllocatorsFromTblInfo(b.Requirement, dbInfo.ID, tblInfo)
 }
 
-func (b *Builder) applyCreateTable(m *meta.Meta, dbInfo *model.DBInfo, tableID int64, allocs autoid.Allocators, tp model.ActionType, affected []int64, schemaVersion int64) ([]int64, error) {
+func applyCreateTable(b *Builder, m *meta.Meta, dbInfo *model.DBInfo, tableID int64, allocs autoid.Allocators, tp model.ActionType, affected []int64, schemaVersion int64) ([]int64, error) {
```
Comment (tiancaiamao) [inline pkg/infoschema/builder.go:632]: It supports both v1 and v2, so renamed according to the name convention.

**Review by GMHDBJD — APPROVED** [review]
LGTM

File: pkg/infoschema/builder.go [hunk 3]
```diff
@@ -792,10 +763,10 @@ func (b *Builder) deleteReferredForeignKeys(dbInfo *model.DBInfo, tableID int64)
 }
 
 // Build builds and returns the built infoschema.
-func (b *Builder) Build() InfoSchema {
+func (b *Builder) Build(schemaTS uint64) InfoSchema {
 	b.updateInfoSchemaBundles(b.infoSchema)
 	if b.enableV2 {
-		b.infoschemaV2.ts = math.MaxUint64 // TODO: should be the correct TS
+		b.infoschemaV2.ts = schemaTS
```
Comment (tiancaiamao) [inline pkg/infoschema/builder.go:769]: The core change of this PR!

**Review by ywqzzy — APPROVED** [review]

**Review by 3pointer — APPROVED** [review]
LGTM

**Pull request merged**